### PR TITLE
Add OMotM and LCotM April

### PR DESCRIPTION
**Pull request by KrisXV**
@TheImmortal

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -364,32 +364,49 @@ let Formats = [
...
 
-		mod: 'gen7',
-		gameType: 'doubles',
-		// searchShow: false,
-		teamLength: {
-			validate: [2, 4],
-			battle: 2,
+		mod: 'linked',
+		ruleset: ['Gen 7] OU'],
+		banlist: ['King\'s Rock', 'Razor Fang'],
```
Comment (Zarel) [inline config/formats.js]: We use double quotes for names just so these don't need to be escaped.

File: config/formats.js [hunk 2]
```diff
@@ -364,32 +364,49 @@ let Formats = [
...
 		],
 
-		mod: 'gen7',
-		gameType: 'doubles',
-		// searchShow: false,
-		teamLength: {
-			validate: [2, 4],
-			battle: 2,
+		mod: 'linked',
+		ruleset: ['Gen 7] OU'],
```
Comment (Zarel) [inline config/formats.js]: `[`

File: config/formats.js (hunk 1 above)
Comment (KrisXV) [reply config/formats.js]: Since when? I've never seen that anywhere in `formats.js`

File: config/formats.js (hunk 2 above)
Comment (KrisXV) [reply config/formats.js]: Why do I always do this, I did it with STABmons too when it was first re-implemented

File: config/formats.js [hunk 3]
```diff
@@ -549,11 +567,28 @@ let Formats = [
...
+		],
+
+		mod: 'gen7',
+		gameType: 'doubles',
+		searchShow: false,
+		teamLength: {
+			validate: [2, 4],
+			battle: 2,
+		},
+		ruleset: ['Gen 7] Doubles OU'],
```
Comment (Marty-D) [inline config/formats.js]: Also here, but this one was apparently around all last month.

Comment (KrisXV) [reply config/formats.js]: ok well thanks.

File: mods/linked/scripts.js [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+
+		this.eachEvent('Update');
+
+		return false;
+	},
+
+	pokemon: {
+		moveUsed: function (move, targetLoc) {
+			let lastMove = this.moveThisTurn ? [this.moveThisTurn, this.battle.getMove(move).id] : this.battle.getMove(move).id;
+			this.lastMove = lastMove;
```
Comment (Slayer95) [inline mods/linked/scripts.js]: ``pokemon.lastMove`` should be a ``Move`` instance now I think?

File: mods/linked/statuses.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+				id: 'confused',
+				effectType: 'Move',
+				type: '???',
+			});
+			return false;
+		},
+	},
+
+	/**
+	 * Gems and Auras
```
Comment (Slayer95) [inline mods/linked/statuses.js]: This comment should be updated now that ``aura`` is no longer here.

**Comment by Slayer95** [comment]
Glad to see Linked back!

My main concerns over porting it to Gen 7 were Dancer and Instruct. I see Dancer is already handled somehow; however, Instruct is not mentioned anywhere. How does it behave? Is it specified in the Gen 7 Linked thread or up to implementation?

File: mods/linked/moves.js [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+				return null;
+			}
+			if (target.volatiles.mustrecharge && target.volatiles.mustrecharge.duration < 2) {
+				// Duration may not be lower than 2 if Sucker Punch is used as a low-priority move
+				// i.e. if Sucker Punch is linked with a negative priority move
+				this.attrLastMove('[still]');
+				this.add('-fail', source);
+				return null;
+			}
+			if (!target.linked && action.move.category === 'Status' && action.move.id !== 'mefirst') {
```
Comment (Slayer95) [inline mods/linked/moves.js]: I don't think ``target.linked`` exists. For reference, this is from my XY implementation:

```js
	suckerpunch: {
		inherit: true,
		onTry: function (source, target) {
			let decision = this.willMove(target);
			if (!decision || decision.choice !== 'move') {
				this.add('-fail', source);
				return false;
			}
			if (target.volatiles['mustrecharge'] && target.volatiles['mustrecharge'].duration < 2) {
				// Duration may not be lower than 2 if Sucker Punch is used as a low-priority move
				// i.e. if Sucker Punch is linked with a negative priority move
				this.add('-fail', source);
				return false;
			}
			if (!decision.linked) {
				if (decision.move.category !== 'Status' || decision.move.id === 'mefirst') return;
				this.add('-fail', source);
				return false;
			}

			for (let i = 0; i < decision.linked.length; i++) {
				let linkedMove = this.getMove(decision.linked[i]);
				if (linkedMove.category !== 'Status' || linkedMove.id === 'mefirst') return;
			}
			this.add('-fail', source);
			return false;
		},
	},
```

So it should be ``action.linked`` now I guess?

File: mods/linked/moves.js [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+				disabled: false,
+				used: false,
+				virtual: true,
+			};
+			this.add('-start', source, 'Mimic', move.name);
+		},
+	},
+
+	/**
+	 * Copycat and Mirror Move
```
Comment (Slayer95) [inline mods/linked/moves.js]: What happened to Copycat?

File: mods/linked/moves.js [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+			case 2:
+				return 60;
+			case 3:
+				return 50;
+			default:
+				return 40;
+			}
+		},
+	},
+};
```
Comment (Slayer95) [inline mods/linked/moves.js]: Is Fusion Bolt + Fusion Flare no longer supported?
Or does it work correctly with default mechanics now?

File: mods/linked/statuses.js (hunk 5 above)
Comment (KrisXV) [reply mods/linked/statuses.js]: oops

File: mods/linked/moves.js (hunk 6 above)
Comment (KrisXV) [reply mods/linked/moves.js]: it was target.linked, idk why it got changed

File: mods/linked/moves.js (hunk 7 above)
Comment (KrisXV) [reply mods/linked/moves.js]: removed because the code was unnecessary

File: mods/linked/moves.js (hunk 8 above)
Comment (KrisXV) [reply mods/linked/moves.js]: it works correctly

File: mods/linked/moves.js (hunk 6 above)
Comment (Slayer95) [reply mods/linked/moves.js]: Probably a bug then. Apply the patch pls :P
There is another bad instance of ``target.linked`` below too.

File: mods/linked/moves.js (hunk 8 above)
Comment (KrisXV) [reply mods/linked/moves.js]: just kidding, after testing, they dont work

File: mods/linked/scripts.js (hunk 4 above)
Comment (KrisXV) [reply mods/linked/scripts.js]: yep

**Comment by KrisXV** [comment]
alright @Slayer95, i believe I fixed everything. I also spoke with that ivy guy and he just said to make Instruct work like other calling/copying moves, so I just adjusted it for ``getLastMoveAbsolute``. (also i fixed the stupid comma dangle thing)

**Comment by VineST** [comment]
I don't remember ever getting specific with Instruct, but it makes sense to have it respect the last absolute move if there were to ever be a Doubles context.

File: mods/linked/scripts.js [hunk 9]
```diff
@@ -0,0 +1,508 @@
...
+		}
+
+		this.eachEvent('Update');
+
+		return false;
+	},
+
+	pokemon: {
+		moveUsed: function (move, targetLoc) {
+			let lastMove = this.moveThisTurn ? [this.moveThisTurn, move] : move;
```
Comment (Slayer95) [inline mods/linked/scripts.js]: I don't think this actually affects gameplay, but ``this.moveThisTurn`` is a string, while ``move`` is a ``Move`` instance, so this is a weird tuple.

If you are interested, you might want to normalize the types.

**Comment by Slayer95** [comment]
Sorry for the late response. I just want to point out that this looks great, except for a tiny typing nit, noted in a comment above.

**Pull request merged**